enterprise/frontend/src/embedding-sdk/types.ts

```diff
@@ -1,6 +1,35 @@
+import type { SerializedError } from "@reduxjs/toolkit";
+
+import type { State } from "metabase-types/store";
+
+type SDKAuthType =
+  | {
+      authType: "apiKey";
+      apiKey: string;
+    }
+  | {
+      authType: "jwt";
+      jwtProviderUri: string;
+    };
+
 export type SDKConfigType = {
   metabaseInstanceUrl: string;
   font?: string;
-  authType: "apiKey";
-  apiKey: string;
+} & SDKAuthType;
+
+export type EmbeddingSessionTokenState = {
+  token: {
+    id: string;
+    exp: number;
+  } | null;
+  loading: boolean;
+  error: SerializedError | null;
 };
+
+export interface EnterpriseState extends State {
+  plugins: {
```

Comment:
I'm wondering if we should put `embeddingSessionToken` directly in the state without this extra nested layer

Reply:
that was something i've been thinking about over the weekend. maybe i'll ask on the thread